src/statements.md

```diff
@@ -28,29 +32,29 @@ const userTypeLiteral = new UserType {
   bar = 3
 }
 const arrayTypeLiteral = new Array<int64> [ 1, 1, 2, 3, 5, 8, 13 ]
-const mapTypeLiteral = new Map<bool, string> {
+const otherArrayTypeLiteral = [ "foo", "bar", "baz" ]
+const mapTypeLiteral = new HashMap<bool, string> {
   true: "It's true!"
   false: "Nuh uh!"
 }
 ```
 
-All three complex type literals begin with the `new` keyword, followed by the type name, which must be a realized generic if it is a generic type. If the original generic type was an `Array` or `Map` it gets a special syntax. Arrays are surrounded by brackets (`[`, `]`) instead of curly braces (`{`, `}`) and are simply a comma-delimited list of values (they could also be generated by functions or operator expressions, or be a user-defined type literal). Maps are keep the curly braces, but the `key` and `value` pairs are separated by a colon (`:`) instead of an equal sign (`=`), which fits more closely to the Map/Object syntax popularized by JSON (but able to have more than just `string` values for the keys).
+All three complex type literals begin with the `new` keyword, followed by the type name, which must be a realized generic if it is a generic type. If the original generic type was an `Array` or `HashMap` it gets a special syntax. Arrays are surrounded by brackets (`[`, `]`) instead of curly braces (`{`, `}`) and are simply a comma-delimited list of values (they could also be generated by functions or operator expressions, or be a user-defined type literal). Arrays are also the only special type that can be unambiguously declared without the `new Type` prefix, so that is also allowed. HashMaps keep the curly braces, but the `key` and `value` pairs are separated by a colon (`:`) instead of an equal sign (`=`), which fits more closely to the Map/Object syntax popularized by JSON (but able to have more than just `string` values for the keys).
```

Comment:
down the line it would be nice to offer the same short notation for `HashMap` so 
```
const mapTypeLiteral = new HashMap<bool, string> {
  true: "It's true!"
  false: "Nuh uh!"
}
```
can simply be

```
const mapTypeLiteral = {
  true: "It's true!"
  false: "Nuh uh!"
}
```

Reply:
I don't know if that can be done. I would like it to be the case, but since these things use `{}` as anonymous side-effect-only function definitions, I don't know if the ambiguity would be a problem:

```ln
on start {
  print('This is a callback function that takes no argument and returns nothing!')
  emit exit 0
}
```

```ln
if a == b {
  print('This is also an anonymous function taking no argument and returning no value')
} else {
  print('Same here')
}
```

Reply:
Fair point